container/docker/docker_test.go

```diff
@@ -0,0 +1,237 @@
+	cn "github.com/docker/docker/api/types/container"
+	nw "github.com/docker/docker/api/types/network"
+	"github.com/docker/docker/client"
+	. "github.com/smartystreets/goconvey/convey"
+	"github.com/wtsi-ssg/wr/container"
+)
+
+const (
+	testReaderCloserStats = `{
+		"read":"2021-01-05T11:42:54.959351591Z",
+		"preread":"2021-01-05T11:42:53.949728039Z",
+		"pids_stats":{"current":4},
+		"blkio_stats":{},
+		"num_procs":0,
+		"storage_stats":{},
+		"cpu_stats":{
+			"cpu_usage":{
+				"total_usage":1244741231366,
+				"percpu_usage":[924236203020,320505028346],
+				"usage_in_kernelmode":9190000000,
+				"usage_in_usermode":653150000000
+			},
+			"system_cpu_usage":2053540000000,
+			"online_cpus":2,
+			"throttling_data":{"periods":0,"throttled_periods":0,"throttled_time":0}
+		},
+		"precpu_stats":{},
+		"memory_stats":{
+			"usage":57921536,
+			"max_usage":115904512,
+			"stats":{
+				"active_anon":1216512,
+				"active_file":41766912,
+				"cache":53268480,
+				"dirty":135168,
+				"hierarchical_memory_limit":9223372036854771712,
+				"hierarchical_memsw_limit":9223372036854771712,
+				"inactive_anon":0,
+				"inactive_file":11354112,
+				"mapped_file":3514368,
+				"pgfault":97911,
+				"pgmajfault":165,
+				"pgpgin":66198,
+				"pgpgout":52876,
+			    "rss":1048576,
+				"rss_huge":0,
+				"total_active_anon":1216512,
+				"total_active_file":41766912,
+				"total_cache":53268480,
+				"total_dirty":135168,
+				"total_inactive_anon":0,
+				"total_inactive_file":11354112,
+				"total_mapped_file":3514368,
+				"total_pgfault":97911,
+				"total_pgmajfault":165,
+				"total_pgpgin":66198,
+				"total_pgpgout":52876,
+				"total_rss":1048576,
+				"total_rss_huge":0,
+				"total_unevictable":0,
+				"total_writeback":0,
+				"unevictable":0,
+				"writeback":0},
+				"limit":2084458496
+			},
+			"name":"/test_container2",
+			"id":"container_id2",
+			"networks":{}
+		}`
+)
+
+// createContainers creates and starts the test containers, given a list of container names.
+func createContainers(ctx context.Context, cli *client.Client, containerNames []string) error {
+	for _, cname := range containerNames {
+		// create the docker container
+		cbody, err := cli.ContainerCreate(ctx, &cn.Config{Image: "ubuntu", Tty: true}, &cn.HostConfig{},
+			&nw.NetworkingConfig{}, cname)
+		if err != nil {
+			return err
+		}
+
+		// start the docker container
+		err = cli.ContainerStart(ctx, cbody.ID, types.ContainerStartOptions{})
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// removeContainers removes all the containers given a list of containers as a part of clean up step.
+func removeContainers(ctx context.Context, cli *client.Client, containerList []*container.Container) error {
+	for _, cntr := range containerList {
+		err := cli.ContainerRemove(ctx, cntr.ID, types.ContainerRemoveOptions{Force: true})
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func TestDockerFuncs(t *testing.T) {
```

Comment:
Just "TestDocker". No need to include "funcs". We know we're always going to be testing functions.

Reply:
Done

Reply:
Added another test to create a docker operator.

Also, modified existing tests to take "pre-existing containers" into consideration. It's a bit hacky but it should just do fine.
- create a slice of dummy containerIDs after the creation. 
- check the length of the list of current containers to be greater than or equal to the length of containerIDs slice

The reason for the 'greater than or equal to' check was, because we may have pre-existing containers and any of those containers may get removed while our tests are running so I cannot have a variable holding their count as it may change in between.